### PR TITLE
Cone positions and color to ros

### PR DESCRIPTION
**Pull request by davidoort**
This PR achieves the following:
image (4)

by adding a couple (yellow and blue) append cone events and methods to the Referee blueprint which get called in the splines cones blueprint at begin play. The Referee now stores the car initial position in Unreal Engine coordinates and the blue and yellow cone locations which are sent through Rpc and can be extracted with the airsim client method getRefereeState which is what is done in the airsim ros wrapper constructor.

I would really appreciate if this could be tested in windows

### CODE REVIEW

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp [hunk 1]
```diff
@@ -131,9 +200,13 @@ void AirsimROSWrapper::create_ros_pubs_from_settings_json()
         set_nans_to_zeros_in_pose(*vehicle_setting);
 
         vehicle_name = curr_vehicle_name;
-        odom_pub = nh_.advertise<nav_msgs::Odometry>("odom", 10);
+        odom_pub = nh_.advertise<nav_msgs::Odometry>("for_testing/odom", 10);
```
Comment (SijmenHuizenga) [inline ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:203]: not in line with docs

File: docs/ros-bridge.md [hunk 2]
```diff
@@ -29,9 +29,12 @@ Velocity, orientation and acceleratoin information at 250hz.
 [Read all about the IMU model here.](imu.md)
 Data is in the `fsds/FSCar` frame.
 
-- `/fsds/odom` [nav_msgs/Odometry](https://docs.ros.org/api/nav_msgs/html/msg/Odometry.html)   
+- `/fsds/for_testing/odom` [nav_msgs/Odometry](https://docs.ros.org/api/nav_msgs/html/msg/Odometry.html)
```
Comment (SijmenHuizenga) [inline docs/ros-bridge.md:32]: inconsistent

File: AirSim/AirLib/include/vehicles/car/api/CarRpcLibAdapators.hpp [hunk 3]
```diff
@@ -80,21 +82,57 @@ class CarRpcLibAdapators : public RpcLibAdapatorsBase {
...
+
     struct RefereeState {
         int doo_counter = 0;
-	std::vector<float> laps;
+	    std::vector<float> laps;
+        std::vector<Cone> cones;
+        Point2D initial_position;
 
-        MSGPACK_DEFINE_MAP(doo_counter, laps);
+        MSGPACK_DEFINE_MAP(doo_counter, laps, cones, initial_position, initial_position);
```
Comment (SijmenHuizenga) [inline AirSim/AirLib/include/vehicles/car/api/CarRpcLibAdapators.hpp:104]: 2x initial_position

File: docs/ros-bridge.md [hunk 4]
```diff
@@ -29,9 +29,12 @@ Velocity, orientation and acceleratoin information at 250hz.
 [Read all about the IMU model here.](imu.md)
 Data is in the `fsds/FSCar` frame.
 
-- `/fsds/odom` [nav_msgs/Odometry](https://docs.ros.org/api/nav_msgs/html/msg/Odometry.html)   
+- `/fsds/for_testing/odom` [nav_msgs/Odometry](https://docs.ros.org/api/nav_msgs/html/msg/Odometry.html)   
 Ground truth car position and orientation in NED frame. THIS WILL NOT BE STREAMED DURING COMPETITION.
 
+- `/fsds/track` [fsds_ros_bridge/Track](https://github.com/FS-Online/Formula-Student-Driverless-Simulator/blob/master/ros/src/fsds_ros_bridge/msg/Track.msg)   
+Ground truth cone position and color wrt starting location of the car. THIS WILL NOT BE STREAMED DURING COMPETITION.
```
Comment (SijmenHuizenga) [inline docs/ros-bridge.md:36]: what is `wrt`?

Can document that this only publishes the cones that are part of the track spline? Any cones placed manually in the world are not published here.

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp [hunk 5]
```diff
@@ -131,9 +200,13 @@ void AirsimROSWrapper::create_ros_pubs_from_settings_json()
         set_nans_to_zeros_in_pose(*vehicle_setting);
 
         vehicle_name = curr_vehicle_name;
-        odom_pub = nh_.advertise<nav_msgs::Odometry>("odom", 10);
+        odom_pub = nh_.advertise<nav_msgs::Odometry>("for_testing/odom", 10);
         global_gps_pub = nh_.advertise<sensor_msgs::NavSatFix>("gps", 10);
         imu_pub = nh_.advertise<sensor_msgs::Imu>("imu", 10);
+        // TODO: remove track publisher at competition
+        track_pub = nh_.advertise<fsds_ros_bridge::Track>("for_testing/track", 10, true);
+        viz_track_pub = nh_.advertise<visualization_msgs::MarkerArray>("for_testing/viz/track", 1, true);
```
Comment (SijmenHuizenga) [inline ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:208]: undocumented

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp [hunk 6]
```diff
@@ -75,6 +75,72 @@ void AirsimROSWrapper::initialize_statistics()
...
+        } else if (cone.color == CarApiBase::ConeColor::OrangeSmall) {
+            cone_object.color = fsds_ros_bridge::Cone::ORANGE_SMALL;
+        } else if (cone.color == CarApiBase::ConeColor::Unknown) {
+            cone_object.color = fsds_ros_bridge::Cone::UNKNOWN;
+        }
+        track.track.push_back(cone_object);
+        
+        // Fill Marker
+        visualization_msgs::Marker cone_marker;
+        // Note that because choosing this NED frame, Rviz will display a map which is flipped wrt the real map
```
Comment (SijmenHuizenga) [inline ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:106]: let's also add this in the docs

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp [hunk 7]
```diff
@@ -75,6 +75,72 @@ void AirsimROSWrapper::initialize_statistics()
...
+        // Note that because choosing this NED frame, Rviz will display a map which is flipped wrt the real map 
+        cone_marker.header.frame_id = "fsds/FSCar";
+        cone_marker.header.stamp = ros::Time::now();
+        cone_marker.lifetime = ros::Duration();
+        cone_marker.id = cone_id;
+        cone_marker.type = visualization_msgs::Marker::SPHERE;
+        cone_marker.action = visualization_msgs::Marker::ADD;
+        cone_marker.pose.position.x = cone_object.location.x;
+        cone_marker.pose.position.y = cone_object.location.y;
+        cone_marker.pose.position.z = 0;
```
Comment (SijmenHuizenga) [inline ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:115]: I understand that it is _easy_ to add the visualisation topics in here, but I'm not sure this is the right place. If we would have endless time I would split all the visualisation things into their own node. But we are on a time budget so let's keep this for now.

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp [hunk 8]
```diff
@@ -99,6 +166,8 @@ void AirsimROSWrapper::initialize_ros()
 
     statistics_timer_ = nh_private_.createTimer(ros::Duration(1), &AirsimROSWrapper::statistics_timer_cb, this);
     go_signal_timer_ = nh_private_.createTimer(ros::Duration(1), &AirsimROSWrapper::go_signal_timer_cb, this);
+
+    publish_track();
```
Comment (SijmenHuizenga) [inline ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:170]: Sooo the track is published once? Is this desired behavior? Can we document this?

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 7 above)
Comment (davidoort) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:115]: so should I keep this here for now? Honestly this visualization code/ topic is really only here for debugging so you can replicate my results and quickly see that the track has the correct shape. I am in favor of removing this code once you approve

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 5 above)
Comment (davidoort) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:208]: what part? the viz marker?

File: AirSim/AirLib/include/vehicles/car/api/CarRpcLibAdapators.hpp (hunk 3 above)
Comment (davidoort) [reply AirSim/AirLib/include/vehicles/car/api/CarRpcLibAdapators.hpp:104]: good catch

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 1 above)
Comment (davidoort) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:203]: I can't find a place where odom is mentioned without the prefix

File: docs/ros-bridge.md (hunk 2 above)
Comment (davidoort) [reply docs/ros-bridge.md:32]: with what?

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 5 above)
Comment (SijmenHuizenga) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp]: yes

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 1 above)
Comment (SijmenHuizenga) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp]: the inconsistency was that some were `for_testing` and some `testing`. Thanks for fixing it!

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 7 above)
Comment (SijmenHuizenga) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp:115]: Yea, let's remove it then :) I think it is amazing to see the cones in rviz but I don't think it should be the responsibility of the ros bridge to provide these topics.

File: ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp (hunk 5 above)
Comment (SijmenHuizenga) [reply ros/src/fsds_ros_bridge/src/airsim_ros_wrapper.cpp]: well if we are deleting it then don't bother ;)

**Pull request merged**